### PR TITLE
DNM/WIP: Manifest v3 Migration (Firefox Preview) 

### PR DESCRIPTION
**Pull request by maxxcrawford**
- WIP - Manifest v3 Migration
- Update build command to use Firefox Preview

### CODE REVIEW

**Comment by willdurand** [comment]
You also need to update `"web_accessible_resources"` in the `manifest.json` file. Something like that, although you should probably have a more strict set of URL patterns in `"matches"`:

```diff
diff --git a/src/manifest.json b/src/manifest.json
index 46f8f99..68c11cd 100644
--- a/src/manifest.json
+++ b/src/manifest.json
@@ -91,12 +91,17 @@
   ],
 
   "web_accessible_resources": [
-    "images/*.svg",
-    "icons/*.svg",
-    "icons/*.png",
-    "fonts/Inter/*.woff2",
-    "fonts/Metropolis/*.woff2",
-    "*.html"
+    {
+      "resources": [
+        "images/*.svg",
+        "icons/*.svg",
+        "icons/*.png",
+        "fonts/Inter/*.woff2",
+        "fonts/Metropolis/*.woff2",
+        "*.html"
+      ],
+      "matches": ["<all_urls>"]
+    }
   ]
 
 }
```

File: src/manifest.json [hunk 1]
```diff
@@ -1,72 +1,85 @@
...
-      }
+    "gecko": {
+      "id": "private-relay@firefox.com",
+      "strict_min_version": "79.0"
+    }
   },
-
   "background": {
-      "scripts": ["js/libs/browser-polyfill.min.js", "js/background/background.js", "js/background/context-menu.js"]
+    "service_worker": "js/background/background.js",
```
Comment (willdurand) [inline src/manifest.json:19]: service worker isn't enabled on Firefox yet, what you want is an event page

**Comment by maxxcrawford** [comment]
Closing this PR as it has served its purpose. We can revisit the manifest file when ready to migrate. Adding a few review notes for some code changes that I'm not 100% sure why was made (by me). 😎

File: src/js/background/background.js [hunk 2]
```diff
@@ -7,7 +7,7 @@ browser.storage.local.set({ relayApiSource: `${RELAY_SITE_ORIGIN}/api/v1` });
 browser.runtime.onInstalled.addListener(async (details) => {
   const { firstRunShown } = await browser.storage.local.get("firstRunShown");
 
-  if (details.reason == "update") {
+  if (details.reason == "update" || details.reason == "install") {
```
Comment (maxxcrawford) [inline src/js/background/background.js:10]: question: Why?

File: src/js/popup/popup.js [hunk 3]
```diff
@@ -266,6 +265,14 @@ async function showRelayPanel(tipPanelToShow) {
   let onboardingPanelStrings = await getOnboardingPanels();
 
   const isBundleAvailableInCountry = (await browser.storage.local.get("bundlePlans")).bundlePlans.BUNDLE_PLANS.available_in_country;
+  // const isBundleAvailableInCountry = (await browser.storage.local.get("bundlePlans")).bundlePlans.BUNDLE_PLANS.available_in_country;
+
+  await browser.runtime.sendMessage({
+    method: "storeRuntimeData",
+    forceUpdate: true
+  });
```
Comment (maxxcrawford) [inline src/js/popup/popup.js:273]: Question: Why?

**Review by maxxcrawford — COMMENTED** [review]
Questions asked. 🤷

**Pull request closed without merging**